test_util/test_aws_cf.py

```diff
@@ -0,0 +1,140 @@
+
+    Raises:
+        AssertionError: if any environment variables or resources are missing
+            or do not conform
+    """
+    options = type('Options', (object,), {})()
+
+    options.variant = os.getenv('DCOS_VARIANT', 'default')
+
+    options.template_url = os.getenv('DCOS_TEMPLATE_URL')
+    assert options.template_url, 'A CF template must be specified!'
+
+    options.ci_flags = os.getenv('CI_FLAGS', '')
+    options.aws_access_key_id = os.getenv('AWS_ACCESS_KEY_ID')
+    options.aws_secret_access_key = os.getenv('AWS_SECRET_ACCESS_KEY')
+    assert options.aws_access_key_id, 'AWS Credentials Required!'
+    assert options.aws_secret_access_key, 'AWS Credentials Required!'
+
+    add_env = {}
+    prefix = 'TEST_ADD_ENV_'
+    for k, v in os.environ.items():
+        if k.startswith(prefix):
+            add_env[k.replace(prefix, '')] = v
+    options.add_env = add_env
+    options.pytest_dir = os.getenv('DCOS_PYTEST_DIR', '/opt/mesosphere/active/dcos-integration-test')
+    options.pytest_cmd = os.getenv('DCOS_PYTEST_CMD', 'py.test -vv '+options.ci_flags)
+    return options
+
+
+def main():
+    options = check_environment()
+
+    random_identifier = ''.join(random.choice(string.ascii_uppercase + string.digits) for _ in range(10))
+    unique_cluster_id = "CF-integration-test-{}".format(random_identifier)
+    log.info("Spinning up AWS CloudFormation with ID: {}".format(unique_cluster_id))
+    bw = test_util.provide_aws.BotoWrapper(
+            region=DEFAULT_AWS_REGION,
+            aws_access_key_id=options.aws_access_key_id,
+            aws_secret_access_key=options.aws_secret_access_key)
+    # TODO(mellenburg): use randomly generated keys this key is delivered by CI or user
+    ssh_key_path = 'default_ssh_key'
+    cf = test_util.provide_aws.DcosCfSimple.create(
+        stack_name=unique_cluster_id,
+        template_url=options.template_url,
+        private_agents=2,
+        public_agents=1,
+        admin_location='0.0.0.0/0',
+        key_pair_name='default',
+        boto_wrapper=bw)
+    cf.wait_for_stack_creation()
+
+    log.info("Exporting ssh_key to disk as")
+    # key must be chmod 600 for test_runner to use
+    os.chmod(ssh_key_path, stat.S_IREAD | stat.S_IWRITE)
+
+    # Create custom SSH Runnner to help orchestrate the test
+    ssh_user = 'core'
+    remote_dir = '/home/core'
+
+    master_ips = cf.get_master_ips()
+    public_agent_ips = cf.get_public_agent_ips()
+    private_agent_ips = cf.get_private_agent_ips()
+    test_host = master_ips[0][1]
+    log.info('Running integration test from: '+test_host)
+    master_list = [i[0] for i in master_ips]
+    log.info('Master private IPs: '+repr(master_list))
+    agent_list = [i[0] for i in private_agent_ips]
+    log.info('Private agent private IPs: '+repr(agent_list))
+    public_agent_list = [i[0] for i in public_agent_ips]
+    log.info('Public agent private IPs: '+repr(public_agent_list))
```

Comment:
Can we print to the screen what ssh key to use here for developers + the command (assuming a specific default key path), that a developer can use to get into the cluster?